pkg/controllers/nodeclaim/expiration/controller.go

```diff
@@ -72,9 +78,14 @@ func (c *Controller) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim) (re
 }
 
 func (c *Controller) Register(_ context.Context, m manager.Manager) error {
-	builder := controllerruntime.NewControllerManagedBy(m)
-	return builder.
+	return controllerruntime.NewControllerManagedBy(m).
 		Named("nodeclaim.expiration").
-		For(&v1.NodeClaim{}).
+		For(&v1.NodeClaim{}, builder.WithPredicates(predicate.TypedFuncs[client.Object]{
+			CreateFunc: func(_ event.TypedCreateEvent[client.Object]) bool { return true },
+			DeleteFunc: func(_ event.TypedDeleteEvent[client.Object]) bool { return false },
+			UpdateFunc: func(_ event.TypedUpdateEvent[client.Object]) bool { return false },
+			GenericFunc: func(_ event.TypedGenericEvent[client.Object]) bool { return false },
+
+		})).
```

Comment:
I think this is actually harder for us to test and reason about that I'd rather we just do this in the reconcile loop. The computation benefit we get from short circuiting three simple if statements I think is negligible.